src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -392,12 +411,16 @@ namespace winrt::TerminalApp::implementation
 
                 return false;
             };
-            if (ShouldUsePersistedLayout(_settings) && _startupActions.Size() == 1 && firstActionIsDefault(_startupActions.GetAt(0)))
-            {
-                auto layouts = ApplicationState::SharedInstance().PersistedWindowLayouts();
-                if (layouts && layouts.Size() > 0 && layouts.GetAt(0).TabLayout() && layouts.GetAt(0).TabLayout().Size() > 0)
+            const auto idx = LoadPersistedLayoutIdx(_settings);
+            // It is not well defined what should happen if a user specifies a
+            // command line and also to load from a particular layout.
+            if (idx && _startupActions.Size() == 1 && firstActionIsDefault(_startupActions.GetAt(0)))
```

Comment:
should we debug assert that this case isn't actually possible (having a layout and a commandline?)

I suppose it's technically possible for someone to do `wt -s 1 nt ; sp im-a-trouble-maker.exe` but in that case, we'll just ignore the saved layout and instead do `nt ; sp im-a-trouble-maker.exe`, yea?

Seems reasonable

Reply:
Looking back, I haven't actually tested this with custom `startupActions` from the config yet, which presumably would render this expression false. When this was initially implemented I didn't have a `HasNoCommandlineArgs`. Perhaps the decision tree could look like this

- When launching for the first time
    - if the user provided no commandline args and has no startupActions run the first saved layout in the current window
    - Run all other saved layouts in new windows, and if a saved layout idx is provided always use that (overriding any commandline args or startupActions)?

Reply:
I'm kinda leaning towards the "ignore the `-s` arg" variation as it currently is. IMO, `--saved` is an implementation detail that users _shouldn't_ be using themselves. If they give us some other commandline other than the `--saved`, then I'd defer to the commandline

Reply:
Just to confirm, what behavior should I implement if the user has `startupActions` in their settings.json? Currently if a user has custom `startupActions` it will completely override the saved layout implementation, I believe.

Reply:
Oh yikes okay so I'm thinking the order is:
* If we've got a persisted layout arg (and no commandline), load the persisted layout. (ignore any `startupActions`) (this is the `wt -s 1` case that we spawn)
* If we've got a persisted layout arg and a commandline, then just do the commandline. (This is the `wt -s 1 nt ; sp` case that the user really shouldn't be doing)
* If we don't have a persisted layout arg:
  * run the commandline (if there is one) (e.g. `wt nt ; sp`)
  * else do the `startupActions` if that's set
  * else just make a tab with the default profile
  
  
Sorry if I made that more confusing.

Reply:
With some well placed comments documenting assumptions, does the following sound correct
- If there is a saved layout specified, always run that on load. Since for now it is internal only we can ensure that all of our calls have no commandline
- Otherwise, use the normal order of command line -> startupActions -> just a new tab

Reply:
Made it so that if there is a commandline provided or startup actions we will always create new windows for the saved layouts. Then if a saved layout index is specified we will always use that overriding everything else.